src/containers/projects-table/projects-table.tsx

```diff
@@ -3,54 +3,48 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { useEffect, useState } from 'react';
-import { AxiosResponse } from 'axios';
 import { Project, WorkPackage } from 'utils';
-import { apiFetch } from '../../shared/axios';
 import { weeksPipe, fullNamePipe, wbsPipe } from '../../shared/pipes';
 import PrjsTable from '../../components/projects-table/projects-table'; // Directly rename the default import
-import { DisplayProject } from '../../components/projects-table/projects-table';
 import './projects-table.module.css';
+import { useAllProjects } from '../../services/projects';
 
 const ProjectsTable: React.FC = () => {
-  const [allProjects, setAllProjects] = useState<DisplayProject[]>([]); // store projects data
-
-  // Transforms given project data and sets local state
-  const updateData = (response: AxiosResponse) => {
-    setAllProjects(
-      response.data.map((prj: Project) => {
-        return {
-          wbsNum: wbsPipe(prj.wbsNum),
-          name: prj.name,
-          projectLead: fullNamePipe(prj.projectLead),
-          projectManager: fullNamePipe(prj.projectManager),
-          duration: weeksPipe(
-            prj.workPackages.reduce((tot: number, cur: WorkPackage) => tot + cur.duration, 0)
-          )
-        };
-      })
-    );
+  const { isLoading, errorMessage, responseData } = useAllProjects();
+
+  /**
+   * Transform given list of projects into a list of display projects.
+   * @param projects list of projects to transform
+   * @returns list of display projects
+   */
+  const updateData = (projects: Project[]) => {
```

Comment:
I might suggest a better name for this function like `projectsToDisplayProjects()` or `transformToDisplayProjects()`